stages/02_process_data.py

```diff
@@ -0,0 +1,32 @@
+# Imports
+# -------
+import os
+import gzip
+import pandas as pd
+
+_, _, files = next(os.walk('downloads'))
+files = [ os.path.join('downloads', file) for file in files ]
+
+os.makedirs('brick', exist_ok=True)
+
+for file in files:
+
+  out_file = os.path.join('brick', file.split('/')[1].split('.')[0] + '.parquet')
```

Comment:
```python
out_file = brick_dir / f.relative_to(download_dir).with_suffix('.parquet')
```

This makes it more clear what is happening. Also handles paths with a depth greater than one.

Reply:
Hmm, you might have to do something more along the lines of:

```python
brick_dir / f.relative_to(cache_dir).with_name( re.sub( r'\.(txt\.gz|xlsx)$', '.parquet', f.name ) 
```
because `.with_suffix()` can only remove the final suffix component.

Reply:
I think this is a little too complex. It's pretty simple with the Posix Path object I think and easy to scale.

Reply:
If the concern is that the regex is a bit complex to edit, it can be constructed from a list.
```python
import re
extensions = ['txt.gz', 'xlsx']
extensions_re = re.compile(r'\.(' + '|'.join(re.escape(ext) for ext in extensions) + r')$')

# ...

out_basename = re.sub(extensions_re, '.parquet', f.name )
out_file = brick_dir / f.relative_to(download_dir).with_name( out_basename )
```

Notice that the input files each have a `v3.3` before the file suffix? Using `split('.')[0]` removes the `.3` part. This more declarative code is easier to read than a series of `split`s (`split()` assumes too much about how files are named).

Reply:
That was the concern. We can use that for later on as what the file inputs are when querying someone.